File: getWeChat.c
#include <stdio.h>
#include <string.h>
#include <stdlib.h>

char num2char[] = {
    'a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', 'i', 'j',
    'k', 'l', 'm', 'n', 'o', 'p', 'q', 'r', 's', 't',
    'u', 'v', 'w', 'x', 'y', 'z', '_'
};

void int2str(char *input) {
    char cleaned_input[1000];
    int cleaned_len = 0;
    for (int i = 0; i < strlen(input); i++) {
        if (input[i] != ' ') {
            cleaned_input[cleaned_len++] = input[i];
        }
    }
    cleaned_input[cleaned_len] = '\0';
    char result[500];
    int result_len = 0;
    for (int i = 0; i < cleaned_len; i += 2) {
        char num_str[3] = { cleaned_input[i], cleaned_input[i + 1], '\0' };
        int num = atoi(num_str);

        if (num >= 1 && num <= 27) {
            result[result_len++] = num2char[num - 1];
        } else {
            printf("Error!\n");
            return;
        }
    }
    result[result_len] = '\0';
    printf("My WeChat ID is: %s\n", result);
}

int main() {
    char input[1000];
    printf("Enter your answer: ");
    fgets(input, 1000, stdin);
    input[strcspn(input, "\n")] = '\0'; 
    int2str(input);
    return 0;
}

File: getWeChat.py
num2char = {
    1: 'a', 2: 'b', 3: 'c', 4: 'd', 5: 'e', 
    6: 'f', 7: 'g', 8: 'h', 9: 'i', 10: 'j',
    11: 'k', 12: 'l', 13: 'm', 14: 'n', 
    15: 'o', 16: 'p', 17: 'q', 18: 'r', 
    19: 's', 20: 't', 21: 'u', 22: 'v', 
    23: 'w', 24: 'x', 25: 'y', 26: 'z', 
    27: '_'
}

def int2str(_input):
    _input = _input.replace(" ", "")
    result = []
    for i in range(0, len(_input), 2):
        num = int(_input[i: i + 2])
        if num in num2char:
            result.append(num2char[num])
        else:
            print("Error!")
            return
    WeChatID = ''.join(result)
    print("My WeChat ID is:", WeChatID)

your_input = input("Enter your answer: ")
int2str(your_input)

